### PR TITLE
DEV-1074: Additional recording type support 

### PR DESCRIPTION
**Pull request by kimberleejohnson**
This PR adds support for additional Daily recording types.

### CODE REVIEW

File: custom/recording/contexts/RecordingProvider.js [hunk 1]
```diff
@@ -85,11 +92,26 @@ export const RecordingProvider = ({ children }) => {
       }
     };
 
+    // The 'recording-data' event is emitted when an output-byte-stream recording has started
+    const handleRecordingData = async (ev) => {
+      try {
+        console.log('got data', ev);
+        await window.writer.write(ev.data);
```
Comment (philmillman) [inline custom/recording/contexts/RecordingProvider.js:99]: where is window.writer defined?

Comment (jessmitch42) [reply custom/recording/contexts/RecordingProvider.js:99]: I think Phil might have the answer here. In the docs example, we define window.writer in recording-started but it looks like we're missing that here. (haven't tested yet, trying now)

Comment (jessmitch42) [reply custom/recording/contexts/RecordingProvider.js:99]: `
  /**
   * Handle recording started.
   */
  const handleRecordingStarted = useCallback(
    (event) => {
      console.log('RECORDING');
      console.log(event);

      if (recordingState === RECORDING_RECORDING) return;
      setRecordingState(RECORDING_RECORDING);
      if (event.local) {
        // Recording started locally, either through UI or programmatically
        setIsRecordingLocally(true);
        if (!recordingStartedDate) setRecordingStartedDate(new Date());
        if (event.type === 'output-byte-stream') {
          const { readable, writable } = new TransformStream({
            transform: (chunk, ctrl) => {
              chunk.arrayBuffer().then((b) => ctrl.enqueue(new Uint8Array(b)));
            },
          });
          window.writer = writable.getWriter();
          readable.pipeTo(streamSaver.createWriteStream('test-vid.mp4'));
        }
      }
    },
    [recordingState, recordingStartedDate]
  );`
  
  This seemed to resolve the error but test it out to see what you think

Comment (kimberleejohnson) [reply custom/recording/contexts/RecordingProvider.js:99]: Whoops! Thank you for catching this, @jessmitch42 and @philmillman. I've updated and all seems to be working. I can see the `finished: true` status in the event object, but is there any other best practice we should share in the docs or demo to help customers retrieve the recording? Thanks again.

finished_true

**Review by philmillman — APPROVED** [review]
🚀

**Review by lizadaily — APPROVED** [review]
lgtm!

**Review by ablwr — APPROVED** [review]

**Pull request merged**